VenafiPS/Public/Add-TppAdaptableHash.ps1

```diff
@@ -0,0 +1,175 @@
+
+    .PARAMETER Class
+    Used when setting policy attributes for an Adaptable App. The only valid parameter is 'Adaptable App'.
+
+    .PARAMETER Keyname
+    The name of the Secret Encryption Key (SEK) to used when encrypting this item. Default is "Software:Default"
+
+    .PARAMETER FilePath
+    Required. The full path to the adaptable script file. This should normally be in a 
+    '<drive>:\Program Files\Venafi\Scripts\<subdir>' directory for TPP to recognize the script.
+
+    .PARAMETER VenafiSession
+    Authentication for the function.
+    The value defaults to the script session object $VenafiSession created by New-VenafiSession.
+    A TPP token or VaaS key can also provided.
+    If providing a TPP token, an environment variable named TPP_SERVER must also be set.
+
+    .INPUTS
+    Path and FilePath
+
+    .OUTPUTS
+    None
+
+    .EXAMPLE
+    Add-TppAdaptableHash -Path $Path -Class 'Adaptable App' -FilePath 'C:\Program Files\Venafi\Scripts\AdaptableApp\AppDriver.ps1'
+
+    Update the hash on an adaptable app object.
+
+    .EXAMPLE
+    Add-TppAdaptableHash -Path $Path -FilePath 'C:\Program Files\Venafi\Scripts\AdaptableLog\Generic-LogDriver.ps1'
+
+    Update the hash on an adaptable log object.
+
+    .LINK
+    http://VenafiPS.readthedocs.io/en/latest/functions/Add-TppAdaptableHash/
+
+    .LINK
+    https://github.com/Venafi/VenafiPS/blob/main/VenafiPS/Public/Add-TppAdaptableHash.ps1
+
+    .LINK
+    https://docs.venafi.com/Docs/current/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-add.php
+
+    .LINK
+    https://docs.venafi.com/Docs/currentSDK/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-ownerdelete.php
+
+    .LINK
+    https://docs.venafi.com/Docs/current/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-retrieve.php
+    #>
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript( {
+                if ( $_ | Test-TppDnPath ) {
+                    $true
+                } else {
+                    throw "'$_' is not a valid DN path"
+                }
+            })]
+        [Alias('DN')]
+        [String] $Path,
+
+        [Parameter()]
+        [ValidateSet('Adaptable App')]
+        [ValidateNotNullOrEmpty()]
+        [string] $Class,
+
+        [Parameter()]
+		[string] $Keyname = "Software:Default",
+
+        [Parameter(Mandatory)]
+        [Alias('File')]
+        [string] $FilePath,
+
+        [Parameter()]
+        [psobject] $VenafiSession = $VenafiSession
+    )
+
+    begin {
+        Test-VenafiSession -VenafiSession $VenafiSession -Platform 'TPP'
+
+        $params = @{
+            VenafiSession = $VenafiSession
+            Method        = 'Post'
+        }
+
+        if ( $Class ) {
+            $retrieveVaultID = ( Get-TppAttribute -Path $Path -Class $Class -Attribute 'PowerShell Script Hash Vault Id' ).'PowerShell Script Hash Vault Id'
```

Comment:
as a shortcut, if you want, you can use -AsValue to get right to the value for 1 attribute
```suggestion
            $retrieveVaultID = Get-TppAttribute -Path $Path -Class $Class -Attribute 'PowerShell Script Hash Vault Id' -AsValue
```